sklearn/cross_validation.py

```diff
@@ -259,6 +259,10 @@ def __init__(self, n, n_folds, indices, shuffle, random_state):
         if not isinstance(shuffle, bool):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
+
+        if random_state is not None and not shuffle:
+            warnings.warn('shuffle must be True if you need a non default random_state.')
```

Comment:
I feel this could be written as -

    `shuffle` should be set to True for the data to be shuffled based on the given `random_state`.